flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSourceITCase.java

```diff
@@ -572,7 +572,8 @@ public void testStreamPartitionReadByPartitionName() throws Exception {
                 .commit("pt_year='2020',pt_mon='09',pt_day='03'");
 
         TableResult result =
-                tEnv.executeSql("select * from hive.source_db.stream_partition_name_test");
+                tEnv.executeSql(
+                        "select * from hive.source_db.stream_partition_name_test where pt_year = '2020'");
```

Comment:
I think this test can pass without this patch, we should change to  `select  x, y from...` thus the selected fields push down.

Reply:
Hi, the test does not pass when add a `where` condition, it got a `TestTimedOutException` , and it pass with the patch and no exception throws.  So would we set the test sql like  `select x,y from .... where ....` ?  @leonardBang

Reply:
> the test does not pass when add a where condition, it got a TestTimedOutException

Do we know why is that? It sounds like another hidden issue to me.

Reply:
From the debug log, it is the same problem: partition key field not found. the projected fields do not include the partition column both in `select x, y from ..` and `selecrt * from .. where` .

Reply:
I see. Since we filter partition by a literal, the partition value doesn't need to be selected from the source. Thanks for the explanations.